libMBIN/Source/Template/NMSTemplate.cs

```diff
@@ -657,10 +658,16 @@ public void SerializeGenericList( BinaryWriter writer, IList list, long listHead
                 int alignment_default = 0x4;
                 if (list[0].GetType().BaseType == typeof(NMSTemplate)) {
```

Comment:
this might actually be written a lot more cleanly as a switch statement, and then we can check for multiple things with the same alignment in one line I think? Ie. we don't need to check for UInt16 in one else if and Int16 in another else if

Reply:
It would be better as a switch.  In other places in the code where is does a switch on a type it uses the type name.  I could do the same here.  You can't switch on the type itself becuase typeof(type) is not a constant expression.

Reply:
huh, interesting, I did not know that!